tools/create_nifti_bval_bvec.pl

```diff
@@ -0,0 +1,201 @@
+       $0 -help to list options
+USAGE
+
+&Getopt::Tabular::SetHelp($Help, $Usage);
+&Getopt::Tabular::GetOptions(\@opt_table, \@ARGV)
+    || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+
+
+
+## input error checking
+
+if (!$ENV{LORIS_CONFIG}) {
+    print STDERR "\n\tERROR: Environment variable 'LORIS_CONFIG' not set\n\n";
+    exit $NeuroDB::ExitCodes::INVALID_ENVIRONMENT_VAR;
+}
+
+if (!defined $profile || !-e "$ENV{LORIS_CONFIG}/.loris_mri/$profile") {
+    print $Help;
+    print STDERR "$Usage\n\tERROR: You must specify a valid and existing profile.\n\n";
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+
+if ( !@Settings::db ) {
+    print STDERR "\n\tERROR: You don't have a \@db setting in the file "
+        . "$ENV{LORIS_CONFIG}/.loris_mri/$profile \n\n";
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+
+
+## establish database connection
+
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+print "\n==> Successfully connected to the database \n" if $verbose;
+
+
+
+## get config settings
+
+my $data_dir   = NeuroDB::DBI::getConfigSetting(\$dbh, 'dataDirBasepath');
+my $create_nii = NeuroDB::DBI::getConfigSetting(\$dbh, 'create_nii');
+
+
+
+## exit if create_nii is set to No
+
+unless ($create_nii) {
+    print "\nConfig option 'create_nii' set to no. bvec/bval files will not be "
+          . "created as NIfTI files are not created by the imaging pipeline\n\n";
+    exit $NeuroDB::ExitCodes::SUCCESS;
+}
+
+
+
+## grep all FileIDs for which acquisition:bvalues are set
+
+print "\n==> Fetching all FileIDs with acquisition:bvalues. \n" if $verbose;
+
+( my $query = <<QUERY ) =~ s/\n/ /g;
+  SELECT
+    FileID
+  FROM
+    parameter_file
+    JOIN parameter_type USING (ParameterTypeID)
+  WHERE
+    parameter_type.Name=?
+QUERY
+
+my $sth = $dbh->prepare($query);
+$sth->execute('acquisition:bvalues');
+
+my @file_ids;
+while( my $fileID = $sth->fetchrow_array) {
+    push @file_ids, $fileID;
+}
+
+unless (@file_ids) {
+    print "\n No files were found with header 'acquisition:bvalues' so no need "
+          . "to create bval/bvec files \n";
+    exit $NeuroDB::ExitCodes::SUCCESS;
+}
+
+
+
+## loop through all FileIDs and create bvec/bval files
+foreach my $file_id (@file_ids) {
+
+    # load the file based on the FileID
+    my $file = NeuroDB::File->new(\$dbh);
+    $file->loadFile($file_id);
+
+    # determine paths for bval/bvec files
+    my $minc = $file->getFileDatum('File');
+    my ($bval_file, $bvec_file) = map { $minc }(0 .. 1);
+    $bval_file =~ s/mnc$/bval/g;
+    $bvec_file =~ s/mnc$/bvec/g;
+
+    # create complementary nifti files for DWI acquisitions
+    my $bval_success = NeuroDB::MRI::create_dwi_nifti_bval_file(
+        \$file, $data_dir . '/' . $bval_file
+    );
+    my $bvec_success = NeuroDB::MRI::create_dwi_nifti_bvec_file(
+        \$file, $data_dir . '/' . $bvec_file
+    );
+    if ($bval_success && $bvec_success) {
```

Comment:
Is is possible that one file creation succeeds and the other one fails? Should the `if` be split to consider this case?